hadoop-common-project/hadoop-common/src/main/java/io/hops/security/ServiceJWTManager.java

```
@@ -0,0 +1,592 @@
+        CommonConfigurationKeysPublic.DEFAULT_JWT_MANAGER_SERVICE_INVALIDATE_PATH);
+    if (!serviceJWTInvalidatePathConf.endsWith("/")) {
+      serviceJWTInvalidatePathConf = serviceJWTRenewPathConf + "/";
+    }
+    serviceJWTInvalidatePath = new URL(host, serviceJWTInvalidatePathConf);
+    super.serviceInit(conf);
+  }
+  
+  private void createLockFileIfMissing() throws IOException {
+    Path lockFile = getLockFilePath();
+    if (!lockFile.toFile().exists()) {
+      FileChannel fc = FileChannel.open(lockFile, StandardOpenOption.CREATE, StandardOpenOption.WRITE);
+      FileLock fl = fc.tryLock();
+      if (fl == null) {
+        return;
+      }
+      try {
+        Files.setPosixFilePermissions(lockFile, LOCK_FILE_PERMISSIONS);
+      } finally {
+        fl.release();
+      }
+    } else {
+      Files.setPosixFilePermissions(lockFile, LOCK_FILE_PERMISSIONS);
+    }
+  }
+  
+  private Path getLockFilePath() {
+    String tmp = System.getProperty("java.io.tmpdir");
+    return Paths.get(tmp, LOCK_FILE);
+  }
+  
+  @Override
+  protected void serviceStart() throws Exception {
+    createLockFileIfMissing();
+    tokenRenewer.execute(new TokenRenewer());
+    super.serviceStart();
+  }
+  
+  @Override
+  protected void serviceStop() throws Exception {
+    try {
+      tokenRenewer.shutdown();
+      if (!tokenRenewer.awaitTermination(100L, TimeUnit.MILLISECONDS)) {
+        tokenRenewer.shutdownNow();
+      }
+    } catch (InterruptedException ex) {
+      tokenRenewer.shutdownNow();
+    }
+    
+    waiter.await();
+    // Try to delete lock file if there is no lock
+    Path lockFile = getLockFilePath();
+    if (lockFile.toFile().exists()) {
+      FileChannel fc = FileChannel.open(lockFile, StandardOpenOption.WRITE);
+      FileLock fl = null;
+      try {
+        fl = fc.tryLock();
+        if (fl != null) {
+          FileUtils.deleteQuietly(lockFile.toFile());
+        }
+      } catch (Exception ex) {
+        // Oo well
```

should we not at least print a warning?
And also have a special case for interrupt exceptions that would not swallow the exception.

---

Sure, I can print a warning.

Regarding the InterruptedException, since the process is already in a stopping phase, hence the `serviceStop()` is being called, I'm not sure if I should re-throw it. Maybe I can also print a warning that it was interrupted while waiting as they do in `ContainerManagerImpl#cleanUpApplicationsOnNMShutDown` for example

---

Sonarqube will not like that :p